manic/sourcetree.py

```diff
@@ -351,3 +352,18 @@ def checkout(self, verbosity, load_all, load_comp=None):
         # now give each external an opportunitity to checkout it's externals.
         for comp in load_comps:
             self._all_components[comp].checkout_externals(verbosity, load_all)
+
+    def order_comps_by_local_path(self, comps_in):
+        """
+        put the comps into an order so that comp local_paths
+        that are nested are checked out in correct order
+        """
+        comps_out = []
+        local_paths = []
+        for comp in comps_in:
+            local_paths.append(self._all_components[comp].get_local_path())
+        for path in sorted(local_paths, key=len):
+            for comp in comps_in:
+                if self._all_components[comp].get_local_path() == path:
+                    comps_out.append(comp)
+        return comps_out
```

Comment:
I believe this entire function could be replaced by the following (assuming you go with my other suggestion of sorting based on the path rather than the string length... if you want to sort based on the string length, this would just need a minor tweak) (UNTESTED, except via somewhat-similar code in an interactive python session):

```python
load_comps = sorted(tmp_comps, key=lambda comp: self._all_components[comp].get_local_path())
```

Reply:
Thanks!